config/config.go

```diff
@@ -92,62 +112,149 @@ func (c Config) validateUnrecognizedParameters(params Parameters) []error {
-		}
-	case ParameterTypeDuration:
-		_, err := time.ParseDuration(value)
-		if err != nil {
-			return fmt.Errorf("error validating %q: %q value is not a duration: %w", key, value, ErrInvalidParameterType)
+	var errs []error
+	for _, k := range keys {
+		value := c[k]
+		// empty value is valid for all types
+		if value == "" {
+			continue
 		}
-	case ParameterTypeBool:
-		_, err := strconv.ParseBool(value)
-		if err != nil {
-			return fmt.Errorf("error validating %q: %q value is not a boolean: %w", key, value, ErrInvalidParameterType)
+		//nolint:exhaustive // type ParameterTypeFile and ParameterTypeString don't need type validations (both are strings or byte slices)
+		switch param.Type {
+		case ParameterTypeInt:
+			_, err := strconv.Atoi(value)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error validating %q: %q value is not an integer: %w", k, value, ErrInvalidParameterType))
+			}
+		case ParameterTypeFloat:
+			_, err := strconv.ParseFloat(value, 64)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error validating %q: %q value is not a float: %w", k, value, ErrInvalidParameterType))
+			}
+		case ParameterTypeDuration:
+			_, err := time.ParseDuration(value)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error validating %q: %q value is not a duration: %w", k, value, ErrInvalidParameterType))
+			}
+		case ParameterTypeBool:
+			_, err := strconv.ParseBool(value)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error validating %q: %q value is not a boolean: %w", k, value, ErrInvalidParameterType))
+			}
 		}
 	}
-	return nil
+	return errors.Join(errs...)
 }
 
 // validateParamValue validates that a configuration value matches all the
 // validations required for the parameter.
 func (c Config) validateParamValue(key string, param Parameter) error {
-	value := c[key]
 	var errs []error
+	for _, k := range c.getKeysForParameter(key) {
+		value := c[k]
+		var valErrs []error
 
-	isRequired := false
-	for _, v := range param.Validations {
-		if _, ok := v.(ValidationRequired); ok {
-			isRequired = true
+		isRequired := false
+		for _, v := range param.Validations {
+			if _, ok := v.(ValidationRequired); ok {
+				isRequired = true
+			}
+			err := v.Validate(value)
+			if err != nil {
+				valErrs = append(valErrs, fmt.Errorf("error validating %q: %w", k, err))
+				continue
+			}
 		}
-		err := v.Validate(value)
-		if err != nil {
-			errs = append(errs, fmt.Errorf("error validating %q: %w", key, err))
-			continue
+		if value == "" && !isRequired {
+			continue // empty optional parameter is valid
 		}
-	}
-	if value == "" && !isRequired {
-		return nil // empty optional parameter is valid
+		errs = append(errs, valErrs...)
 	}
 
 	return errors.Join(errs...)
 }
 
+func (c Config) getKeysForParameter(key string) []string {
+	// First break up the key into tokens.
+	tokens := strings.Split(key, "*")
```

Comment:
It shouldn't start with a wildcard, we should validate that, you're right 👍

Reply:
I realized that `getKeysForParameter` actually happily accepts a wildcard at the start of the pattern, so I just added tests to make sure it works as expected. I currently don't see when we'd use it, but I also don't see a reason to prevent us from doing that at some point 🤷